sample/tips.cr

```diff
@@ -12,58 +12,37 @@ require "../src/router"
 # - HTTP::StaticFileHandler         https://crystal-lang.org/api/HTTP/StaticFileHandler.html
 # - HTTP::WebSocketHandler          https://crystal-lang.org/api/HTTP/WebSocketHandler.html
 
-class WebServer
+struct WebServer
   include Router
 
-  # HTTP::Handler(s)
-  @log_handler = HTTP::LogHandler.new(STDOUT)
-  @error_handler = HTTP::ErrorHandler.new
-  @static_file_handler = HTTP::StaticFileHandler.new(File.expand_path("../public", __FILE__))
-
-  def initialize
-  end
-
-  def draw_routes
+  def routes
     get "/" do |context, params|
       context.response.print "OK"
       context
     end
   end
-
-  def run
-    # Drawing routes for this server
-    draw_routes
-
-    # Please note about the order of the handlers.
-    # 1. LogHandler should be the first of the array
-    #    since it should get all accesses.
-    # 2. ErrorHandler should be the next of the LogHandler to handle all errors.
-    # 3. StatiFileHandler should be the next of the ErrorHandler
-    #    since it should serve static files before accesses coming to RouteHandler.
-    #    StaticFileHandler will pass the access to the RouteHandler
-    #    if the file or directory does not exist.
-    # 4. So RouteHandler should be last.
-    # 
-    # The array of the handlers should be like this.
-    # Note: if a route can't be handled by RouteHandler (a.k.a. route not found)
-    # and this handler is the last, a 404 error response will be returned;
-    # otherwise the execution will continue with the next handler in a row
-    handlers = [
-      @log_handler,
-      @error_handler,
-      @static_file_handler,
-      route_handler,
-    ]
-
-    server = HTTP::Server.new(3000, handlers)
-    server.listen
-
-    # Try
-    # `curl localhost:3000/ok`         <- Handle route
-    # `curl localhost:3000/hello.html` <- Serve static file
-    # `curl localhost:3000/invalid`    <- Internal Error(500)
-  end
 end
 
-web_server = WebServer.new
-web_server.run
+# HTTP::Handler(s)
+# Please note about the order of the handlers.
+# 1. LogHandler should be the first of the array
+#    since it should get all accesses.
+# 2. ErrorHandler should be the next of the LogHandler to handle all errors.
+# 3. StatiFileHandler should be the next of the ErrorHandler
+#    since it should serve static files before accesses coming to RouteHandler.
+#    StaticFileHandler will pass the access to the RouteHandler
+#    if the file or directory does not exist.
+# 4. So RouteHandler should be last.
+#
+# The array of the handlers should be like this.
+# Note: if a route can't be handled by RouteHandler (a.k.a. route not found)
+# and this handler is the last, a 404 error response will be returned;
+# otherwise the execution will continue with the next handler in a row
+
+web_server = WebServer.new.run(handlers: [HTTP::LogHandler.new(STDOUT),
```

Comment:
What do you think about having the handlers in the class constructor (`WebServer.new(handlers: [...]).run)`? I've tried to do so but haven't succeed.